### PR TITLE
Implement user in Comments

### PR DESCRIPTION
**Pull request by deciocandido**

### CODE REVIEW

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 1]
```diff
@@ -21,6 +24,15 @@
...
     private String contentText;
 
+    @ManyToOne
+    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "name")
+    @JsonIdentityReference(alwaysAsId = true)
+    @JoinColumn(nullable = false)
+    private User relatedUser;
+    
+    @OneToMany(mappedBy = "likedPost", cascade = CascadeType.ALL)
+    private List<PostLike> listOfCommentLikes;
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java]: Were we not going to have a separate CommentLike model?

Comment (NeuralAlchemist) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java]: Agree, PostLike can only have a list of Post type. It cannot store Comment type.

Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java]: Yes we will have seperate CommentLike model when we implement the likes for comment

File: src/main/java/se/kth/sda/skeleton/postlikes/PostLikeController.java [hunk 2]
```diff
@@ -28,4 +25,15 @@ public PostLikeController(PostLikeService postLikeService) {
...
+
+    /**
+     * Handler method for endpoint "/posts/{postId}/likes" with DELETE HttpRequest. Invokes a HTTP {@link ResponseStatus} of
+     * NO_CONTENT. Deletes the like made by the logged in user for post given by {@code postId}
+     * @param postId the post for which the user's like will be removed
+     */
+    @DeleteMapping("/{postId}/likes")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void removeLike(@PathVariable Long postId){
+        postLikeService.removeLike(postId);
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/postlikes/PostLikeController.java:37]: should we add .orElseThrow(ResourceNotFoundException::new);

File: src/main/java/se/kth/sda/skeleton/user/User.java [hunk 3]
```diff
@@ -16,9 +18,15 @@
...
 
+    @OneToMany(mappedBy = "relatedUser", cascade = CascadeType.ALL)
+    List<Comment> createdComments;
+
     @OneToMany(mappedBy = "likedUser", cascade = CascadeType.ALL)
     List<PostLike> likedPosts;
 
+    @OneToMany(mappedBy = "likedUser", cascade = CascadeType.ALL)
+    List<PostLike> likedComments;
+
```
Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/user/User.java:29]: Please remove this we will implement later

File: src/main/java/se/kth/sda/skeleton/user/User.java [hunk 4]
```diff
@@ -94,4 +102,20 @@ public void setCreatedPosts(List<Post> createdPosts) {
...
+        this.likedComments = likedComments;
+    }
+
+    public List<Comment> getCreatedComments() {
+        return createdComments;
+    }
+
+    public void setCreatedComments(List<Comment> createdComments) {
+        this.createdComments = createdComments;
+    }
```
Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/user/User.java:120]: Please remove this we will implement later

File: src/main/java/se/kth/sda/skeleton/user/User.java (hunk 3 above)
Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/user/User.java]: Please remove this we will implement later

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 5]
```diff
@@ -55,4 +67,20 @@ public Post getRelatedPost() {
...
+        this.relatedUser = relatedUser;
+    }
+
+    public List<PostLike> getListOfLikes() {
+        return listOfCommentLikes;
+    }
+
+    public void setListOfLikes(List<PostLike> listOfCommentLikes) {
+        this.listOfCommentLikes = listOfCommentLikes;
+    }
```
Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java]: Please remove this we will implement later

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 1 above)
Comment (NeuralAlchemist) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java]: Please remove this we will implement later

File: src/main/java/se/kth/sda/skeleton/user/User.java [hunk 6]
```diff
@@ -94,4 +102,20 @@ public void setCreatedPosts(List<Post> createdPosts) {
...
         this.likedPosts = likedPosts;
     }
+
+    public List<PostLike> getLikedComments() {
+        return likedComments;
+    }
+
+    public void setLikedComments(List<PostLike> likedComments) {
+        this.likedComments = likedComments;
+    }
```
Comment (eeels22) [inline src/main/java/se/kth/sda/skeleton/user/User.java]: Please remove this we will implement later

**Review by eeels22 — CHANGES_REQUESTED** [review]
Hi Decio, thanks for the updates. As discussed, please can you remove the code related to postLike with Comments and implement those in a different branch. Many thanks

File: src/main/java/se/kth/sda/skeleton/user/User.java (hunk 3 above)
Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/user/User.java]: it's been removed

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 5 above)
Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java]: it's been removed

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 1 above)
Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java]: it's been removed

File: src/main/java/se/kth/sda/skeleton/user/User.java (hunk 6 above)
Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/user/User.java]: it's been removed

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 7]
```diff
@@ -21,6 +23,12 @@
     @NotBlank
     private String contentText;
 
+    @ManyToOne
+    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "name")
```
Comment (janko101) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java]: Does it send only the name or the whole user?

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java [hunk 8]
```diff
@@ -29,6 +37,8 @@
     @JsonIgnore
     private Post relatedPost;
 
+    LocalDateTime createdTime;
```
Comment (janko101) [inline src/main/java/se/kth/sda/skeleton/comments/Comment.java:40]: So we'll also send a date created?

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 7 above)
Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java]: It sends the whole user

File: src/main/java/se/kth/sda/skeleton/comments/Comment.java (hunk 8 above)
Comment (deciocandido) [reply src/main/java/se/kth/sda/skeleton/comments/Comment.java:40]: Yes for Post and Comment

**Review by eeels22 — APPROVED** [review]

**Pull request merged**